### PR TITLE
Nucleus data sanity

### PR DESCRIPTION
**Pull request by sippycoder**

### CODE REVIEW

File: f_lite/data.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+        logging.info("Data partition begins!")
+        start_time = time.time()  # Record the start time
+        
+        data = []
+        if self.root_dir_type == "parquet":
+            logging.info(f"Loading data from local parquet files: {self.root_dir}")
+            parquet_files = glob.glob(os.path.join(self.root_dir, self.collection_name, "*/*/*.parquet"))
+            for file in tqdm(parquet_files, desc=f"Loading data"):
+                df = pd.read_parquet(file)
+                df = df[df["media_source"] != "laion"]
```
Comment (sippycoder) [inline f_lite/data.py:145]: Laion needs to be included after @nmnWithNucleus cleans the dead links

File: f_lite/model.py [hunk 2]
```diff
@@ -14,6 +14,13 @@
 from torch import nn
 
 
+def kaiming_init(m):
+    if isinstance(m, (nn.Linear, nn.Conv2d)):
+        nn.init.kaiming_normal_(m.weight)
+        if m.bias is not None:
+            nn.init.zeros_(m.bias)
+
+
```
Comment (sippycoder) [inline f_lite/model.py:23]: This initialization is ok? It's compatible with `µP` for later experiments.

File: f_lite/train.py [hunk 3]
```diff
@@ -395,15 +303,15 @@ def encode_prompt_with_t5(
...
-    prompt_embeds = prompt_embeds.hidden_states[return_index]
-    if return_index != -1:
-        prompt_embeds = text_encoder.encoder.final_layer_norm(prompt_embeds)
-        prompt_embeds = text_encoder.encoder.dropout(prompt_embeds)
-
+    if return_index == -1 or return_index == len(text_output.hidden_states) - 1:
+        prompt_embeds = text_output.last_hidden_state
+    else:
+        prompt_embeds = text_output.hidden_states[return_index]
+
```
Comment (sippycoder) [inline f_lite/train.py:314]: This change needs to be reverted? Removing `final_layer_norm` and `dropout` gave gibberish runs. Study `T5EncoderModel`.
